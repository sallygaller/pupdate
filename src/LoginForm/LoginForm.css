.LoginForm {
    display: flex;
    flex-direction: column;
    text-align: left;
    margin: 0 auto;
    width: 30%;
}

.LoginForm input {
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    display: inline-block;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
  }

.LoginForm button {
    width: 100%;
    background-color: #4CAF50;
    color: white;
    padding: 14px 20px;
    margin: 8px 0;
    border: none;
    border-radius: 4px;
    cursor: pointer;
}

.LoginForm button:enabled:hover {
    background-color: #45a049;
}

.red {
    color: red;
}