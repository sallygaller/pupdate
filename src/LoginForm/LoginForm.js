import React from "react";
import TokenService from "../services/token-service";
import AuthApiService from "../services/auth-api-service";
import "./LoginForm.css";

class LoginForm extends React.Component {
  static defaultProps = {
    onLoginSuccess: () => {},
  };

  state = {
    error: null,
  };

  handleSubmitJwtAuth = (e) => {
    e.preventDefault();
    const { email, password } = e.target;

    AuthApiService.postLogin({
      email: email.value,
      password: password.value,
    })
      .then((res) => {
        email.value = "";
        password.value = "";
        TokenService.saveAuthToken(res.authToken);
        this.props.onLoginSuccess();
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <form className="LoginForm" onSubmit={(e) => this.handleSubmitJwtAuth(e)}>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <label htmlFor="email">Email: </label>
        <input
          type="text"
          name="email"
          id="email"
          defaultValue="helenlabrador@madeup.com"
          required
        />
        <label htmlFor="password">Password: </label>
        <input
          type="password"
          name="password"
          id="password"
          defaultValue="Secret12345!"
          required
        />
        <button type="submit">Log In!</button>
      </form>
    );
  }
}

export default LoginForm;
