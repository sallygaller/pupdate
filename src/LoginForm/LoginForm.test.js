import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import LoginForm from "./LoginForm";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <LoginForm />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer.create(<LoginForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
