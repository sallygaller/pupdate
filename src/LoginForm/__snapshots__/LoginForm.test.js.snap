// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <label
    htmlFor="email"
  >
    Email: 
  </label>
  <input
    defaultValue="helenlabrador@madeup.com"
    id="email"
    name="email"
    required={true}
    type="text"
  />
  <label
    htmlFor="password"
  >
    Password: 
  </label>
  <input
    defaultValue="Secret12345!"
    id="password"
    name="password"
    required={true}
    type="password"
  />
  <button
    type="submit"
  >
    Log In!
  </button>
</form>
`;
